services/ns_net/url_loader_impl.mm

```diff
@@ -11,85 +11,96 @@
 
 #import <Foundation/Foundation.h>
 
-@interface URLLoaderConnectionDelegate : NSObject<NSURLConnectionDataDelegate>
+static mojo::URLResponsePtr MojoErrorResponse(NSURL* url, NSString* message) {
+  mojo::URLResponsePtr response = mojo::URLResponse::New();
+  response->url = url.absoluteString.UTF8String;
+  response->error = mojo::NetworkError::New();
```

Comment:
Foundation does not give us a status code here. This is an application level error handling routine instead of a network error.